compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -548,6 +548,22 @@ impl<'a> Parser<'a> {
         is_present
     }
 
+    fn check_noexpect(&self, tok: &TokenKind) -> bool {
```

Comment:
can you double check if the parser has other `self.token ==` use cases that can be turned into `check_noexpect`?